### PR TITLE
Doc update for URI class

### PR DESCRIPTION
**Pull request by jpsiitonen**
Examples added, main page class description added.

### CODE REVIEW

File: sdk/lib/core/uri.dart [hunk 1]
```diff
@@ -161,6 +222,20 @@ abstract class Uri {
   ///
   /// This constructor is the same as [Uri.http] except for the scheme
   /// which is set to `https`.
+  ///
+  /// ```dart
+  /// var uri = Uri.https('example.org', '/path', {'q': 'dart'});
+  /// print(uri); // https://example.org/path?q=dart
+  ///
+  /// uri = Uri.https('user:pass@localhost:8080', '');
```
Comment (alexlindroos) [inline sdk/lib/core/uri.dart]: nit: pass => password ?

File: sdk/lib/core/uri.dart [hunk 2]
```diff
@@ -26,10 +26,73 @@ const String _hexDigits = "0123456789ABCDEF";
...
+/// ## HTTP and HTTPS URI
+/// To create a URI with https scheme, use [Uri.https] or [Uri.http]:
+/// ```dart
+/// final httpsUri = Uri.https('example.com', 'api/fetch', {'limit': '10'});
+/// print(httpsUri); // https://example.com/api/fetch?limit=10
+/// ```
+/// ## File URI
+/// To create a URI from file path, use [Uri.file]:
+/// ```dart
+/// final fileWindows = Uri.file('C:/data/images/image.png', windows: true);
```
Comment (alexlindroos) [inline sdk/lib/core/uri.dart]: nit: all the other examples has windows example second, can we add that second also in this example?

File: sdk/lib/core/uri.dart [hunk 3]
```diff
@@ -582,23 +680,24 @@ abstract class Uri {
   ///
   /// Example:
   /// ```dart
-  /// Uri uri1 = Uri.parse("a://b@c:4/d/e?f#g");
-  /// Uri uri2 = uri1.replace(scheme: "A", path: "D/E/E", fragment: "G");
-  /// print(uri2);  // prints "a://b@c:4/D/E/E?f#G"
+  /// final uri1 = Uri.parse('a://b@c:4/d/e?f#g');
```
Comment (alexlindroos) [inline sdk/lib/core/uri.dart]: Any chance for more clearer example? Like data/images

File: sdk/lib/core/uri.dart [hunk 4]
```diff
@@ -582,23 +680,24 @@ abstract class Uri {
...
   ///     userInfo: uri1.userInfo,
   ///     host: uri1.host,
   ///     port: uri1.port,
-  ///     path: "D/E/E",
+  ///     path: 'D/E/E',
   ///     query: uri1.query,
-  ///     fragment: "G");
-  /// print(uri3);  // prints "a://b@c:4/D/E/E?f#G"
-  /// print(uri2 == uri3);  // prints true.
+  ///     fragment: 'G');
```
Comment (alexlindroos) [inline sdk/lib/core/uri.dart]: Any chance for more clearer example? like data/images

**Review by alexlindroos — CHANGES_REQUESTED** [review]

File: sdk/lib/core/uri.dart (hunk 4 above)
Comment (jpsiitonen) [reply sdk/lib/core/uri.dart]: Added new example with real-world uri changes..

**Review by stepthomp — COMMENTED** [review]
A few lines that are highlighted in red (too long) which will need to be shortened. I see no way of shortening those.

**Pull request closed without merging**